src/dialogsearch.cpp

```diff
@@ -144,7 +144,21 @@ void DialogSearch::on_button_search_clicked()
     }
     else if (ui->radio_switch->isChecked())
     {
-        //TODO implement me
+        int switchID = ui->combo_switch->currentData().toInt();
+
+        search_predicate = [switchID](const RPG::EventCommand& com)
+        {
+            switch(com.code)
+            {
+                case Cmd::ControlSwitches:
+                    return (com.parameters[0] != 2 && com.parameters[1] == switchID) ||
+                        (com.parameters[0] == 1 && com.parameters[2] == switchID);
+                case Cmd::ConditionalBranch:
+                    return com.parameters[0] == 0 && com.parameters[1] == switchID;
+                default:
+                    return false;
```

Comment:
Or use different tabs like the original rpg 2k maker does, but I don't like that :/ Dunno.. Maybe different table header? (We just could use anything different than a (widget-)table too, it was just the first thing that was on my mind that could work)